### DISCUSSION TITLE
Does the App work well with Chinese ROMs?

### DISCUSSION THREAD

**Discussion by snti** [yuriykulikov/AlarmClock#379] [Q&A]
I didn't see any open issue related to this here, in case the App works in Chinese roms what is the trick?
I'm trying to show a notification using alarmanager, it works good in my pixel phone, but some of my friends using xiaomi and huawei says that the notification does not run after a few hours, it only runs a few minutes after it is scheduled, but it works good only if they turn off battery optimization.

Regards!

**Comment by yuriykulikov** [comment]
Hello,
here are issues on some phones, I am not sure which ones.

Perhaps I can add this to the app:
https://developer.android.com/reference/android/provider/Settings.html#ACTION_IGNORE_BATTERY_OPTIMIZATION_SETTINGS

**Comment by temanado** [comment]
> I didn't see any open issue related to this here, in case the App works in Chinese roms what is the trick? I'm trying to show a notification using alarmanager, it works good in my pixel phone, but some of my friends using xiaomi and huawei says that the notification does not run after a few hours, it only runs a few minutes after it is scheduled, but it works good only if they turn off battery optimization.
> 
> Regards!

Hello! Have you figured out what the trick is? I ran into a similar problem, but haven't found a solution yet.

**Reply by yuriykulikov** [reply]
I guess the trick is to use Alarmmanager properly and have permissions in the manifest.
